R/cloud.R

```diff
@@ -530,15 +530,20 @@ cloud_email <- function(type = c("broken", "failed"), job_id = cloud_job(), pkg
 #' This is automatically set by [cloud_check()] and only lasts for the current R session.
 #' @param job_id If not `NULL`, sets the active `job_id` to the input.
 #' @export
-cloud_job <- function(job_id = NULL) {
-  if (!is.null(job_id)) {
-    cloud_data$job_id <- job_id
+cloud_job <- function(pkg = ".", job_id = NULL) {
+  cloud_data$job_id <- job_id %||% cloud_data$job_id
+  if (!is.null(cloud_data$job_id)) {
+    return(invisible(cloud_data$job_id))
   }
 
-  if (is.null(cloud_data$job_id)) {
-    stop("No current job, please specify the `job_id` explicitly, or run a job with `cloud_check()`", call. = FALSE)
+  pkg <- pkg_check(pkg)
+  cloud_dirs <- list.dirs(dir_find(pkg, "cloud"), recursive = FALSE)
```

Comment:
I thought this directory was only created after you called `cloud_report()`?

Reply:
yeah it is created after any data is pulled down, now that we are doing this it might be worth creating it in `cloud_check()` though.

Reply:
Yeah, I assumed I'd need to do a bit more work around this. I was developing this first pass PR w.r.t. usethis, which currently has several cloud check subdirs.